rdflib/plugins/parsers/patch.py

```diff
@@ -0,0 +1,150 @@
+        skolemize: bool = False,
+        **kwargs: Any,
+    ) -> Dataset:
+        """
+        Parse inputsource as an RDF Patch file.
+
+        :type inputsource: `rdflib.parser.InputSource`
+        :param inputsource: the source of RDF Patch formatted data
+        :type sink: `rdflib.graph.Dataset`
+        :param sink: where to send parsed data
+        :type bnode_context: `dict`, optional
+        :param bnode_context: a dict mapping blank node identifiers to `~rdflib.term.BNode` instances.
+                              See `.W3CNTriplesParser.parse`
+        """
+        assert sink.store.context_aware, (
+            "RDFPatchParser must be given" " a context aware store."
+        )
+        # type error: Incompatible types in assignment (expression has type "ConjunctiveGraph", base class "W3CNTriplesParser" defined the type as "Union[DummySink, NTGraphSink]")
+        self.sink: Dataset = Dataset(store=sink.store)  # type: ignore[assignment]
+        self.skolemize = skolemize
+
+        source = inputsource.getCharacterStream()
+        if not source:
+            source = inputsource.getByteStream()
+            source = getreader("utf-8")(source)
+
+        if not hasattr(source, "read"):
+            raise ParseError("Item to parse must be a file-like object.")
+
+        self.file = source
+        self.buffer = ""
+        while True:
+            self.line = __line = self.readline()
+            if self.line is None:
+                break
+            try:
+                self.parsepatch(bnode_context)
+            except ParseError as msg:
+                raise ParseError("Invalid line (%s):\n%r" % (msg, __line))
+
+        if self.skolemize:
+            return self.sink
+        else:
+            self.sink = self.sink.de_skolemize()
+            return self.sink  # Dataset is skolemized as part of adding/removing triples
+            # , so de skolemize before returning. NB this is broken by the parent class
+            # (ConjunctiveGraph) which re-skolemizes the dataset.
+
+    def parsepatch(self, bnode_context: Optional[_BNodeContextType] = None) -> None:
+        self.eat(r_wspace)
+        #  From spec: "No comments should be included (comments start # and run to end
+        #  of line)."
+        if (not self.line) or self.line.startswith("#"):
+            return  # The line is empty or a comment
+
+        # if header, transaction, skip
+        operation = self.operation()
+        self.eat(r_wspace)
+
+        if operation == Operation.AddTripleOrQuad:
+            self.add_triple_or_quad()
+        elif operation == Operation.DeleteTripleOrQuad:
+            self.delete_triple_or_quad()
+        elif operation == Operation.AddPrefix:
+            self.add_prefix()
+        elif operation == Operation.DeletePrefix:
+            self.delete_prefix()
+
+    def add_triple_or_quad(self):
+        self.sink.parse(data=self.line, format="nquads", skolemize=True)
```

Comment:
Is this a performance issue? `sink.parse()` will create a whole new instance of a new NQuads parser, each with its own new `Dataset` to parse into, for every line that is added in the rdp file.
Probably not a big deal for small test file, but with patches of hundreds of MB I can see this will become a bottleneck.

Reply:
I originally used parse line and can had some issue, can't recall what - will have another crack.

Reply:
Since this is reusing the existing parse method from the superclass, I'm guessing it doesn't support the `<_:label>` syntax defined in https://afs.github.io/rdf-delta/rdf-patch.html. Probably need to make a note of it somewhere in the docs.

Reply:
I don't believe the RDF Patch stores need to be concerned with skolemizing and de-skolemizing blank nodes and skolem IRIs. Would it be possible to override the `nodeid` method in the superclass to return the blank node verbatim from the input rather than generating a new bnode id?

Reply:
> Since this is reusing the existing parse method from the superclass, I'm guessing it doesn't support the `<_:label>` syntax defined in https://afs.github.io/rdf-delta/rdf-patch.html. Probably need to make a note of it somewhere in the docs.

Sort of - I tested and RDFLib just parses them as URIRefs, so they'll work similar to skolem IRIs in that they won't be document scoped. It will be problematic for any internal operations which they should be blank nodes for though, and any serialisation would need to convert them back to blank nodes.

With your second comment there on skolemisation - I'll give that a go, and also parse any <_:x> as a blank node with x as the node ID.

Reply:
> I don't believe the RDF Patch stores need to be concerned with skolemizing and de-skolemizing blank nodes and skolem IRIs. Would it be possible to override the `nodeid` method in the superclass to return the blank node verbatim from the input rather than generating a new bnode id?

I've made this change + the support for the <:_label> syntax, could you please review again.